### PR TITLE
Jamf protect for Microsoft Sentinel v2.1

### PR DESCRIPTION
**Pull request by txhaflaire**
Required items, please complete
   
   Change(s):
   - Updated Jamf Protect Workbook including Endpoint Telemetry and Network Event Stream
   - Added 3 new Analytic Rule templates

   Reason for Change(s):
   - Mapping Jamf Protect features into Jamf Protect for Microsoft Sentinel
   - Expanding the Jamf Protect for Microsoft Sentinel solution with Analytic Rules for Incident creation

   Version Updated:
   - No
   - 3 new Analytic Rule templates, starting with version 1.0.0

   Testing Completed:
   - Yes

   Checked that the validations are passing and have addressed any issues that are present:
   - Yes

### CODE REVIEW

**Comment by v-spadarthi** [comment]
@txhaflaire :Please resolve the below validation errors
image
Please remove the below content from workbook
image
We couldn't see the changes related to Analytics rules in Main template Please re-create package again and 
please add correctly analytics content in  solutioninput file 
image

**Comment by txhaflaire** [comment]
@v-spadarthi Fixed both workbooks and Analytic Rules. Please review 👍

**Comment by v-spadarthi** [comment]
@txhaflaire : Please fix the Validation errors
image

**Comment by txhaflaire** [comment]
@v-spadarthi @aprakash13 Please review the following.

Screenshot 2022-12-27 at 11 36 24
In this failed validation it seems that Value is not being recognized, but it's actually an existing value. 
Screenshot 2022-12-27 at 11 31 43
It' fails with the fact that the jamfprotect_CL table is not being recognized, but it's because it's a custom table created via the IngestionAPI.

**Review by aprakash13 — CHANGES_REQUESTED** [review]
For the sake of uniformity can we please have the analytic rule template's fields ordered to something similar to this: https://github.com/Azure/Azure-Sentinel/blob/master/Solutions/AristaAwakeSecurity/Analytic%20Rules/HighMatchCountsByDevice.yaml

They seem to be missing the mandatory fields like 
requiredDataConnectors 
   connectorId
   dataTypes 
Could we please include them as well. 

Also, please fix the validation issues that are currently present. Feel free to reach out if help is needed.
Thanks.

**Comment by txhaflaire** [comment]
> For the sake of uniformity can we please have the analytic rule template's fields ordered to something similar to this: https://github.com/Azure/Azure-Sentinel/blob/master/Solutions/AristaAwakeSecurity/Analytic%20Rules/HighMatchCountsByDevice.yaml
> 
> They seem to be missing the mandatory fields like requiredDataConnectors connectorId dataTypes Could we please include them as well.
> 
> Also, please fix the validation issues that are currently present. Feel free to reach out if help is needed. Thanks.

@v-spadarthi @aprakash13 Added the missing mandatory fields and changed the fields order to match the example.
Please review the validation errors as they are not related to the Analytics Rules in this PR.

Thanks in advance!

**Comment by txhaflaire** [comment]
@v-spadarthi @aprakash13 Removed the RequiredDataConnectors as in this case we are not using a Data Connector. (Jamf Protect Cloud forwards data into Microsoft Analytics Workspace using the IngestionAPI.) Do we need to build a Data Connector then?


Please advice on the KQLValidations error

**Comment by txhaflaire** [comment]
@v-spadarthi @aprakash13 Added a valid and working Data Connector as well; please review the KQLValidations as the errors are not related to this particular PR. 2.1.0.zip works as expected when uploading Partner Center and testing it in Preview mode.

**Comment by txhaflaire** [comment]
@v-spadarthi @aprakash13 @v-mchatla tagging all of you to check if there is availability to help out on this PR or either https://github.com/Azure/Azure-Sentinel/pull/7020

**Comment by aprakash13** [comment]
> @v-spadarthi @aprakash13 @v-mchatla tagging all of you to check if there is availability to help out on this PR or either #7020

Thanks for making the changes. Looking into the KQL validation errors.

**Comment by v-atulyadav** [comment]
Hi @aprakash13, any progress on KQL validation errors. Thanks.

**Comment by txhaflaire** [comment]
Hi @aprakash13 @v-dvedak Any progress on the KQLValidation errors? Anything i can do to speed up the process?

**Comment by v-atulyadav** [comment]
Hi @txhaflaire, we're looking for someone who can help us with our KQL validations. Please give us some time to resolve this issue. Thanks.

**Comment by v-atulyadav** [comment]
Hi @txhaflaire, can you please brief us why crowdstrike Maintemplate.json is included in this PR? if not required then delete this file.

image

I would also like to request a sample data set, as the workbook query fails, or you can paste screenshots of the working workbook here

image

**Comment by txhaflaire** [comment]
@v-atulyadav Removed the CrowdStrike Maintemplate. (Accidentally that file maked it into my PR).

Here with the screenshots as attached; there is already sample data located in. https://github.com/Azure/Azure-Sentinel/blob/master/Sample%20Data/JamfProtectExampleData.csv

Screenshot 2023-01-23 at 11 11 41
Screenshot 2023-01-23 at 11 11 37
Screenshot 2023-01-23 at 11 11 34

**Comment by v-atulyadav** [comment]
Hi @txhaflaire, it has been determined that the validation is failing in data connector because a custom blank array is defined. Could you please remove the following blank property from data connector and repackage the solution? 

image

**Comment by txhaflaire** [comment]
> Hi @txhaflaire, it has been determined that the validation is failing in data connector because a custom blank array is defined. Could you please remove the following blank property from data connector and repackage the solution?
> 
> image
> 
> image

Hi @v-atulyadav First of all, thank-you for all your amazing help.
I have removed the blank array, re-created the package. see commit https://github.com/Azure/Azure-Sentinel/pull/6953/commits/69d161e6e9fbcec517106a55fbc0adcc73808c0c


Thanks!

**Comment by v-atulyadav** [comment]
Hi @txhaflaire, could you please check whether we removed or assigned some values to the blank properties in the analytical rule below. Also repackage after this change. Thanks

image

File: Solutions/Jamf Protect/Analytic Rules/JamfProtectAlerts.yaml
Comment (devikamehra) [inline Solutions/Jamf Protect/Analytic Rules/JamfProtectAlerts.yaml:13]: It is mandatory to provide atleast one tactic and relevant technique in an analytical rule

File: Solutions/Jamf Protect/Analytic Rules/JamfProtectAlerts.yaml
Comment (devikamehra) [inline Solutions/Jamf Protect/Analytic Rules/JamfProtectAlerts.yaml:14]: Please update the field name with correct name (relevantTechniques)

File: Solutions/Jamf Protect/Analytic Rules/JamfProtectAlerts.yaml
Comment (devikamehra) [inline Solutions/Jamf Protect/Analytic Rules/JamfProtectAlerts.yaml:21]: Field input_match_facts_s is getting parsed multiple times. Please parse this field once and use the resultant dynamic object to get the various values.

File: Solutions/Jamf Protect/Analytic Rules/JamfProtectAlerts.yaml
Comment (devikamehra) [inline Solutions/Jamf Protect/Analytic Rules/JamfProtectAlerts.yaml:30]: Use has instead of contains

File: Solutions/Jamf Protect/Analytic Rules/JamfProtectAlerts.yaml
Comment (devikamehra) [inline Solutions/Jamf Protect/Analytic Rules/JamfProtectAlerts.yaml:31]: It would be better to filter the results first and then parse the rest of the fields.

**Comment by txhaflaire** [comment]
> Hi @txhaflaire, could you please check whether we removed or assigned some values to the blank properties in the analytical rule below. Also repackage after this change. Thanks
> 
> image

@v-atulyadav That did the trick, to remove those blank pairs. (I could not find this my self in the ARM-TTK error logs, apologise!)

File: Solutions/Jamf Protect/Analytic Rules/JamfProtectAlerts.yaml
Comment (devikamehra) [inline Solutions/Jamf Protect/Analytic Rules/JamfProtectAlerts.yaml:42]: This field is not expected here. Include this field in alertDescriptionFormat.

File: Solutions/Jamf Protect/Analytic Rules/JamfProtectAlerts.yaml
Comment (devikamehra) [inline Solutions/Jamf Protect/Analytic Rules/JamfProtectAlerts.yaml:40]: Analytic_description is better suitable for alert dedscription and not name

File: Solutions/Jamf Protect/Analytic Rules/JamfProtectAlerts.yaml
Comment (devikamehra) [inline Solutions/Jamf Protect/Analytic Rules/JamfProtectAlerts.yaml:43]: [Confirmation] What is the expected value of Tags here? Does this contain MITRE attack tactic without spaces? If not, please use a suitable value here.

File: Solutions/Jamf Protect/Analytic Rules/JamfProtectAlerts.yaml
Comment (devikamehra) [inline Solutions/Jamf Protect/Analytic Rules/JamfProtectAlerts.yaml:51]: Value is getting covered in Alert properties itself. Duplication of content is not recommended.

File: Solutions/Jamf Protect/Analytic Rules/JamfProtectAlerts.yaml
Comment (devikamehra) [inline Solutions/Jamf Protect/Analytic Rules/JamfProtectAlerts.yaml:79]: Remove the fields which are not required.

File: Solutions/Jamf Protect/Analytic Rules/JamfProtectAlerts.yaml
Comment (devikamehra) [inline Solutions/Jamf Protect/Analytic Rules/JamfProtectAlerts.yaml:15]: This is not required here.

File: Solutions/Jamf Protect/Analytic Rules/JamfProtectUnifiedLogs.yaml
Comment (devikamehra) [inline Solutions/Jamf Protect/Analytic Rules/JamfProtectUnifiedLogs.yaml:30]: Name and Description are the same. Please update appropriately.

**Review by devikamehra — CHANGES_REQUESTED** [review]
Please note the changes suggested in first rule are applicable for all three. Kindly make the changes accordingly.

File: Solutions/Jamf Protect/Analytic Rules/JamfProtectAlerts.yaml
Comment (txhaflaire) [reply Solutions/Jamf Protect/Analytic Rules/JamfProtectAlerts.yaml:13]: Hi @devikamehra 

Not each alert is related to a TTP, looking at other Solutions and Analytic Rules, i see they are also not having a tactic populated. Is this a hard requirement?

File: Solutions/Jamf Protect/Analytic Rules/JamfProtectAlerts.yaml
Comment (txhaflaire) [reply Solutions/Jamf Protect/Analytic Rules/JamfProtectAlerts.yaml:43]: Removed alertTacticsColumnName as it results contains spaces and not only MITRE tags.

**Comment by txhaflaire** [comment]
@devikamehra Are you able to provide me with an update?

**Comment by txhaflaire** [comment]
@v-atulyadav Could you review and approve / merge if all reviews have been passed?

File: Solutions/Jamf Protect/Analytic Rules/JamfProtectAlerts.yaml
Comment (devikamehra) [reply Solutions/Jamf Protect/Analytic Rules/JamfProtectAlerts.yaml:13]: Hi @txhaflaire,
It would be great to have atleast one assigned to each rule. We can look at exceptions if any.

File: Solutions/Jamf Protect/Analytic Rules/JamfProtectAlerts.yaml
Comment (devikamehra) [inline Solutions/Jamf Protect/Analytic Rules/JamfProtectAlerts.yaml:31]: Last condition does not have any effect. We can keep only the first two and the default.

File: Solutions/Jamf Protect/Analytic Rules/JamfProtectNetworkThreats.yaml
Comment (devikamehra) [inline Solutions/Jamf Protect/Analytic Rules/JamfProtectNetworkThreats.yaml:20]: True is not a valid value for Severity

File: Solutions/Jamf Protect/Analytic Rules/JamfProtectUnifiedLogs.yaml
Comment (devikamehra) [inline Solutions/Jamf Protect/Analytic Rules/JamfProtectUnifiedLogs.yaml:19]: This condition will always be true as we are assigning a default value of "Informational".

**Review by devikamehra — CHANGES_REQUESTED** [review]

**Comment by txhaflaire** [comment]
@devikamehra Thanks for all your reviewing! i have resolved your latest comments.

**Comment by txhaflaire** [comment]
@devikamehra @v-dvedak @aprakash13 Please provide me with an update. This PR is already running since 22th of December.

**Comment by devikamehra** [comment]
@txhaflaire 
Only one change. Also, just a quick check, you removed tactics in the last commit. Can we add it back? Remember the tactics are MITRE tactics without spaces.

**Comment by txhaflaire** [comment]
> @txhaflaire Only one change. Also, just a quick check, you removed tactics in the last commit. Can we add it back? Remember the tactics are MITRE tactics without spaces.

@devikamehra For JamfProtectAlerts.yaml and JamfProtectNetworkThreats.yaml `alertTacticsColumnName` is populated.

Can we use values from the query in the `tactics`?

**Review by devikamehra — DISMISSED** [review]

**Comment by devikamehra** [comment]
@v-spadarthi @v-atulyadav Please validate if all the fields are getting mapped in deployment.
@txhaflaire Please make sure the package is updated with all the changes before we test.

**Review by v-atulyadav — DISMISSED** [review]

**Review by v-dvedak — APPROVED** [review]

**Review by aprakash13 — APPROVED** [review]
Devika suggested a bunch of suggestions from an analytics standpoint that seems to have been addressed. Approving from an analytics perspective. Feel free to merge if other components look good.

**Review by v-atulyadav — COMMENTED** [review]
Hi @txhaflaire, I would appreciate it if you could include the older version of 2.0.0 in this PR as well. This solution should also be repackaged with 2.0.1 instead of 2.1.0. The rest of the PR looks good.

image

**Comment by txhaflaire** [comment]
@v-atulyadav Alright i will add the 2.0.0 package in the repository. Are we sure we want to use 2.0.1? it includes major updates and added Analytic Rules and a Data Connector, those were not int 2.0.0. If possible 2.1.0 makes more sense, then if there are minor tweaks to the existing package then indeed 2.1.1 or 2.0.1 would make sense

**Comment by v-atulyadav** [comment]
Hi @txhaflaire, I am not able to see 2.0.0 zip file in this PR? please guide me on this.

image

**Comment by txhaflaire** [comment]
@v-atulyadav If you review the files changed for the last commit it will show up - guessing it's not showing up as the file itself has not been changed compared the one on the repo it self.

image

**Comment by v-atulyadav** [comment]
Hi @txhaflaire, thanks for your clarification on above. Can you Please explain why ispreview:true is set in data connector availability, since this property is always set to false in the solution.

image

**Comment by txhaflaire** [comment]
@v-atulyadav Changed isPreview in DataConnector from true to false, updated package.

**Comment by txhaflaire** [comment]
@v-atulyadav Good morning / afternoon / evening! Any other changes required from my side to have the PR merged?

**Review by v-atulyadav — COMMENTED** [review]
Hi @txhaflaire, it has been seen that 2.0.0 is included in the 2.0.1 zip file after I downloaded and extracted its contents. can you please remove 2.0.0 from 2.1.0 zip. Thanks.

image

**Comment by txhaflaire** [comment]
> Hi @txhaflaire, it has been seen that 2.0.0 is included in the 2.0.1 zip file after I downloaded and extracted its contents. can you please remove 2.0.0 from 2.1.0 zip. Thanks.
> 
> image

@v-atulyadav Removed the 2.0.0 archive from 2.1.0. Should be good now!

**Review by v-atulyadav — DISMISSED** [review]

**Comment by txhaflaire** [comment]
@v-atulyadav Thanks for approving - do we need more approvals for having this PR merged? Have a nice day!

**Review by v-atulyadav — COMMENTED** [review]
Hi @txhaflaire, Please revert your changes for the highlighted folders below. This PR should not include these folders. This PR will cause another solution if we merge it with the below highlighted content.

image

**Comment by txhaflaire** [comment]
> Hi @txhaflaire, Please revert your changes for the highlighted folders below. This PR should not include these folders. This PR will cause another solution if we merge it with the below highlighted content.
> 
> image

@v-atulyadav Fixed! sorry for the inconvenience! I guess i might did a wrong copy/paste!

**Review by v-atulyadav — APPROVED** [review]

**Review by v-rucdu — APPROVED** [review]

**Comment by txhaflaire** [comment]
Thanks everybody involved in this PR! appreciate ALL the efforts and help! @v-atulyadav @v-dvedak @aprakash13 @devikamehra @v-spadarthi

**Pull request merged**
